### ISSUE TITLE
SystemPropsKt breaks Gradle Configuration cache

### ISSUE THREAD

**Issue by changusmc** [Kotlin/kotlinx.coroutines#3352]
My team is currently using coroutines to handle shell commands from the JVM using `runBlocking { suspendingExecuteCommand(command, environment, ttl, ttlUnit) }` which eventually calls https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/jvm/src/Debug.kt#L70 and https://github.com/Kotlin/kotlinx.coroutines/blob/5eca49cfcb25acefaf10ec9b997de537c653b47e/kotlinx-coroutines-core/jvm/src/internal/SystemProps.kt#L13

```
Read system property 'kotlinx.coroutines.debug'
Exception
org.gradle.configurationcache.ConfigurationCacheProblemsException: Configuration cache problems found in this build.	
1 problem was found storing the configuration cache.	
- Plugin 'com.my-company.gradle-enterprise-plugin': read system property 'kotlinx.coroutines.debug'	
  See https://docs.gradle.org/7.3.3/userguide/configuration_cache.html#config_cache:requirements:undeclared_sys_prop_read

Caused by: org.gradle.api.InvalidUserCodeException: Read system property 'kotlinx.coroutines.debug'	
at org.gradle.configurationcache.SystemPropertyAccessListener.systemPropertyQueried(SystemPropertyAccessListener.kt:92)	
at org.gradle.internal.classpath.Instrumented.systemProperty(Instrumented.java:83)	
at org.gradle.internal.classpath.Instrumented.systemProperty(Instrumented.java:77)	
at kotlinx.coroutines.internal.SystemPropsKt__SystemPropsKt.systemProp(SystemProps.kt:17)	
at kotlinx.coroutines.internal.SystemPropsKt.systemProp(Unknown Source)	
at kotlinx.coroutines.DebugKt.<clinit>(Debug.kt:70)	
at kotlinx.coroutines.CoroutineContextKt.newCoroutineContext(CoroutineContext.kt:20)	
at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:49)	
at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)	
at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)	
at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)	
```

**Comment by qwwdfsad** [comment]
Unfortunately, we cannot do much and change coroutines to depend on Gradle API in order to make them usable in Gradle plugins with the support of configuration cache.

I suggest turning this error into [a warning](https://docs.gradle.org/7.3.3/userguide/configuration_cache.html#config_cache:usage:ignore_problems) and reporting this problem to Gradle, because there is now no clear path for either library authors or plugin authors to rely on libraries that access system properties during its initialization (which is a widespread practice)

**Comment by changusmc** [comment]
No problem, it seems this is fixed in Gradle 7.4, I was on Gradle 7.3
